### PR TITLE
Loop checkout url in stripe subscription metadata

### PR DESCRIPTION
**Pull request by valentinludu**
### <samp>🤖 Generated by Copilot at aa465b7</samp>

Added loop checkout feature and fixed subscription bug for Stripe webhooks. This improves the user experience and reliability of the app when working with loop products and spaces.

### WHY
In order to speed up the manual process in Stripe, every time an invoice for a subscription is created we generate the loop checkout url and adding it in stripe subscription metadata.

### CODE REVIEW

File: pages/api/v1/webhooks/stripe/index.ts [hunk 1]
```diff
@@ -266,6 +274,53 @@ export async function stripePayment(req: NextApiRequest, res: NextApiResponse):
...
+        const priceId = subscriptionData?.price?.id;
+        const email = (stripeSubscription.customer as Stripe.Customer)?.email as string | undefined | null;
+
+        if (!stripeSubscription.metadata.loopCheckout && priceId) {
+          const loopItems = await getLoopProducts();
+          const loopItem = loopItems.find((product) => product.externalId === priceId);
+
+          if (!loopItem) {
+            log.warn(
+              `Loop item was not found in order to create a loop url checkout in stripe for the price ${priceId} and space ${spaceId}`
```
Comment (mattcasey) [inline pages/api/v1/webhooks/stripe/index.ts:302]: spaceId should be passed in the second argument along with other key ids: `log[method](message, { spaceId, userId, pageId });`

File: pages/api/v1/webhooks/stripe/index.ts [hunk 2]
```diff
@@ -266,6 +274,53 @@ export async function stripePayment(req: NextApiRequest, res: NextApiResponse):
...
+            : `${loopCheckoutUrl}/${loopItem.entityId}/${loopItem.itemId}?&cartEnabled=false&email=${email}&sub=${subscriptionId}`;
+
+          await stripeClient.subscriptions.update(subscriptionId, {
+            metadata: {
+              ...(stripeSubscription.metadata || {}),
+              loopCheckout: loopUrl
+            }
+          });
+
+          log.info(`Loop checkout url was succesfully added in stripe metadata for the space ${spaceId}`);
```
Comment (mattcasey) [inline pages/api/v1/webhooks/stripe/index.ts:318]: move spaceId to metadata

**Review by mattcasey — APPROVED** [review]

File: pages/api/v1/webhooks/stripe/index.ts (hunk 2 above)
Comment (valentinludu) [reply pages/api/v1/webhooks/stripe/index.ts]: Done

File: pages/api/v1/webhooks/stripe/index.ts (hunk 1 above)
Comment (valentinludu) [reply pages/api/v1/webhooks/stripe/index.ts]: Done

**Pull request merged**